lib/objectid.js

```diff
@@ -0,0 +1,12 @@
+var ObjectId;
+
+try {
+  ObjectId = require('mongodb').ObjectId;
+} catch(e) {
+  if (e.code === 'MODULE_NOT_FOUND')
+    throw e
+}
+
```

Comment:
Another thing to consider is the option to allow user-configurable ObjectId types. Like in model-config.json, we allow `{"objectid": "../path/to/your/custom/objectid"}`, or something along those lines.